opa-filter-core/src/main/java/com/bisnode/opa/spring/security/filter/OpaFilter.java

```diff
@@ -15,32 +18,37 @@
 import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 import java.io.IOException;
+import java.util.List;
 
 public class OpaFilter extends GenericFilterBean {
 
     private static final Logger log = LoggerFactory.getLogger(OpaFilter.class);
 
     private final AccessDecider<HttpServletRequest> decider;
+    private final ApplicationEventPublisher eventPublisher;
 
     /**
      * Creates new {@link OpaFilter} instance.
      *
-     * @param decider {@link AccessDecider} to use in {@link OpaFilter#doFilter(ServletRequest, ServletResponse, FilterChain)} method.
+     * @param decider        {@link AccessDecider} to use in {@link OpaFilter#doFilter(ServletRequest, ServletResponse, FilterChain)} method.
+     * @param eventPublisher {@link ApplicationEventPublisher} to publish events when deny occurs
      */
-    public OpaFilter(@NonNull AccessDecider<HttpServletRequest> decider) {
+    public OpaFilter(@NonNull AccessDecider<HttpServletRequest> decider, ApplicationEventPublisher eventPublisher) {
```

Comment:
I think we should additionally allow one-argument constructor, which would disable event publishing. Then, the version would be 1.1.0

Reply:
Sounds good